src/components/common/Builder/FormBuilder/GenerateField.tsx

```diff
@@ -0,0 +1,43 @@
+import { Controller } from "react-hook-form";
+import { Label } from "../../Label";
+import type { FC, ComponentType } from "react";
+import type { Control } from "react-hook-form";
+import type { PropertyMeta } from "../../../../services/hubspot/types";
+
+type Props = {
+    meta: PropertyMeta;
+    Component: ComponentType<Record<string, unknown>>;
+    control: Control<Record<string, unknown>>,
+    value?: unknown;
+};
+
+const GenerateField: FC<Props> = ({ meta, control, Component }) => {
+    const fieldType = meta?.fieldType;
+    const label = meta?.label;
+
+    if (!meta) {
+        // eslint-disable-next-line no-console
+        console.error("FormBuilder: wrong config - block config not found");
+        return null;
+    }
+
+    if (!Component) {
```

Comment:
This will never happen according to the types, this is always a `ComponentType`

Reply:
As per above, all type checks passed when it was a required value, so there is no need to widen it

Reply:
moved these checks to the parent component.